src/main/java/thePackmaster/powers/rimworldpack/MoodPower.java

```diff
@@ -0,0 +1,190 @@
+
+    public MoodPower(AbstractCreature owner, int amount) {
+        super(POWER_ID, NAME, PowerType.BUFF, false, owner, amount);
+
+        canGoNegative = true;
+        if(amount < 0)
+            type = PowerType.DEBUFF;
+        else
+            type = PowerType.BUFF;
+
+        checkForBreaksAndInsps();
+    }
+
+    @Override
+    public void onInitialApplication() {
+        checkForBreaksAndInsps();
+    }
+
+    @Override
+    public void stackPower(int stackAmount)
+    {
+        amount += stackAmount;
+
+        checkForBreaksAndInsps();
+    }
+
+    private void checkForBreaksAndInsps(){
+
+        //NOTE: If I do an action to reduce or increase the amount instead of
+        //just doing it directly, I'd get a stack overflow if someone applies
+        //more than 4 mood at once (e.g by stacking copies of Sanguine)
+        while (amount >= 4)
+        {
+            getInspiration();
+            amount -= 4;
+            addToBot(new WaitAction(.25f));
+        }
+
+        while (amount <= -4)
+        {
+            getMentalBreak();
+            amount += 4;
+            addToBot(new WaitAction(.25f));
+        }
+
+        if (amount == 0)
+            addToTop(new RemoveSpecificPowerAction(owner, owner, ID));
+
+        if(amount < 0)
+            type = PowerType.DEBUFF;
+        else
+            type = PowerType.BUFF;
+
+        updateDescription();
+    }
+
+    public void getInspiration() {
+        BreaksAndInspirations.inspiration(owner);
+    }
+
+
+
+    public void getMentalBreak(){
+        BreaksAndInspirations.mentalBreak(owner);
+    }
+
+    @Override
+    public void updateDescription() {
+        if(amount > 0 )
+            description = DESCRIPTIONS[0] + amount + DESCRIPTIONS[1];
+        else
+            description = DESCRIPTIONS[0]+ amount + DESCRIPTIONS[2];
+    }
+
+
+    public static class BreaksAndInspirations{
+
+        public static void inspiration(AbstractCreature owner){
+            int rand = AbstractDungeon.cardRandomRng.random(2);
+            for(int i = 0; i < 20; ++i)
+                AbstractDungeon.effectsQueue.add(new ShineSparkleEffect(owner.hb.x + (float)Math.random() * owner.hb.width, owner.hb.y + (float)Math.random() * owner.hb.height));
+            Wiz.atb(new TalkAction(true, inspirationStrings.TEXT[rand], 2.5f, 2.5f));
+            switch (rand)
+            {
+                case 1:
+                    Wiz.applyToSelf(new BufferPower(owner, 1));
+                    break;
+                case 2:
+                    Wiz.atb(new DrawCardAction(2));
+                    Wiz.atb(new GainEnergyAction(2));
+                    break;
+                default:
+                    fast(owner);
+            }
+        }
+
+        private static void fast(AbstractCreature owner){
+
+            int rand = new Random().nextInt(10);
+            if(rand == 0)
+                Wiz.atb(new SFXAction(modID + "fast"));
```

Comment:
So one time out of 10 you get an extra effect for this one.

Some people might find that annoying, but as an amusing little easter egg it's probably fine.